docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,147 @@
 title: Executor
 ---
 
-* Runtime Versioning
+The executor is responsible for dispatching and executing calls into the
+Substrate runtime.
 
-* Native VS Wasm Execution
+## Runtime Execution
 
-* Forkless Runtime Upgrades
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly
+(Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm
+binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when
+the runtime is upgraded. The executor determines which version of the runtime to
+use when dispatching and executing calls.
+
+### Execution Strategy
+
+Before runtime execution begins, the Substrate client proposes which runtime
+execution environment should be used. This is controlled by the execution
+strategy, which can be configured for the different parts of the blockchain
+execution process. The strategies are:
+
+- `Native`: Execute with native build (if available, WebAssembly otherwise).
+- `Wasm`: Only execute with the WebAssembly build.
+- `Both`: Execute with both native (where available) and WebAssembly builds.
+- `NativeElseWasm`: Execute with the native build if possible; if it fails, then
+  execute with WebAssembly.
+
+The default execution strategies for the different parts of the blockchain
+execution process are:
+
+- Syncing: `NativeElseWasm`
+- Block Import: `NativeElseWasm`
+- Block Construction: `Wasm`
+- Off-Chain Worker: `Native`
```

Comment:
This seems to contradict what is said on the high-level docs (https://substrate.dev/docs/en/overview/off-chain-workers), where it says that "With a sandboxed Wasm execution environment, off-chain workers are fully secure...". It seems to imply that off-chain workers are always executed with Wasm.

Reply:
This is a good point, I will follow up.

Reply:
The answer I got is:

> The offchain stuff needs to be compatible with both (like the runtime itself). I was told that main use cases for offchain workers will require a lot of performance though, so they might only be run natively. However I was sure that the strategy is NativeElseWasm, I think we can safely change it to be more inclusive, but if we actually introduce some heavy computations it might require the validators to actually run in native only.

So the answer is that we will update the config. Remember that even `Native` setting will run Wasm if Native is not available.